code-challenges/arrayShift/array-shift.test.js

```diff
@@ -0,0 +1,83 @@
+'use strict';
+
+const insertShiftArray = (arr, value) => {
+  if (Array.isArray(arr)) {
+    let result = [];
+    const middleIndex = Math.ceil(arr.length / 2);
+    for (let i = 0; i < arr.length + 1; i++) {
+      if (i === middleIndex) {
+        result[i] = value;
+      } else if (i > middleIndex) {
+        result[i] = arr[i - 1];
+      } else {
+        result[i] = arr[i];
+      }
+    }
+    arr = result;
+    return arr;
+  } else {
+    return null;
```

Comment:
I don't think returning `null` here is useful. The app is likely going to break anyway if you're expecting to receive an array and are returned null. So I would probably throw an error with a message saying why it broke.